src/main/java/org/hibernate/models/internal/dynamic/DynamicAnnotationUsage.java

```diff
@@ -98,15 +110,76 @@ public <V> V setAttributeValue(String name, V value) {
 			values = new HashMap<>();
 		}
 
+		if ( value != null ) {
+
+			//noinspection unchecked
+			return (V) values.put( name, value );
+		}
 		//noinspection unchecked
-		return (V) values.put( name, value );
+		return (V) values;
 	}
 
-	private static <A extends Annotation> Map<String, Object> extractBaselineValues(AnnotationDescriptor<A> annotationDescriptor) {
+	private static <A extends Annotation> Map<String, Object> extractBaselineValues(
+			AnnotationDescriptor<A> annotationDescriptor,
+			AnnotationTarget target,
+			SourceModelBuildingContext context) {
 		final HashMap<String, Object> values = new HashMap<>();
 		for ( AttributeDescriptor<?> attribute : annotationDescriptor.getAttributes() ) {
-			values.put( attribute.getName(), attribute.getAttributeMethod().getDefaultValue() );
+			values.put( attribute.getName(), getDefaultValue( attribute, target, context ) );
 		}
 		return values;
 	}
+
+	private static Object getDefaultValue(
+			AttributeDescriptor<?> attribute,
+			AnnotationTarget target,
+			SourceModelBuildingContext context) {
+		final Object defaultValue = attribute.getAttributeMethod().getDefaultValue();
+		Object annotation = getValue( defaultValue, target, context );
+		if ( annotation != null ) {
+			return annotation;
+		}
+		return defaultValue;
+	}
+
+	private static Object getValue(Object value, AnnotationTarget target, SourceModelBuildingContext context) {
+		if ( value instanceof Annotation annotation ) {
+			try {
+				return extractDynamicAnnotationUsage( annotation, target, context );
+			}
+			catch (InvocationTargetException | IllegalAccessException e) {
+				throw new AnnotationAccessException( "Error accessing default annotation attribute value", e );
+			}
+		}
+		else if ( value.getClass().isArray() ) {
+			return getList( value, target, context );
+		}
+		return null;
+	}
```

Comment:
I would unify the `getValue` method into the `getDefaultValue` for conciseness:
```suggestion
		final Object defaultValue = attribute.getAttributeMethod().getDefaultValue();
		if ( defaultValue instanceof Annotation annotation ) {
			try {
				return extractDynamicAnnotationUsage( annotation, target, context );
			}
			catch (InvocationTargetException | IllegalAccessException e) {
				throw new AnnotationAccessException( "Error accessing default annotation attribute value", e );
			}
		}
		else if ( defaultValue.getClass().isArray() ) {
			return getList( defaultValue, target, context );
		}
		return defaultValue;
	}
```

Reply:
not sure it is possible, `getList` needs to call `getValue`

Reply:
I agree.  It is normalized out since it happens from 2 places.

Reply:
You can just remove the `AttributeDescriptor` from `getDefaultValue()`'s signature and use that in `getList` as well.